vulnerability-analyzer/src/main/java/org/hyades/processor/scanner/snyk/SnykProcessor.java

```diff
@@ -171,6 +173,18 @@ public void analyzeBatch(final List<RetryableRecord<String, ScanTask>> batch) {
             cacheReport(mappedReport);
         }
 
+        PageMeta pageMeta = page.meta();
+        if (pageMeta != null && pageMeta.errors() != null) {
+            for (final MetaError metaError : pageMeta.errors()) {
+                if (metaError.meta() == null || metaError.meta().purl() == null) {
+                    continue;
+                }
+                var purlWithError = metaError.meta().purl();
+                LOGGER.error("Purl '{}' failed with status {} due to error {} : {}",
+                        purlWithError, metaError.status(), metaError.code(), metaError.title());
```

Comment:
If we know that this PURL failed, we have to report this accordingly with a `ScannerResult` and `SCAN_STATUS_FAILED`. You can use the `reportFailure` method for this. Make sure to also remove the respective PURL from `purlRecords`, otherwise we'll report multiple results for the same PURL (see `for` loop below).